experimental/personnames/src/api.rs

```diff
@@ -226,9 +302,31 @@ pub enum FormattingFormality {
 /// Person name formatter options.
 #[derive(Clone, Eq, PartialEq, Debug)]
 pub struct PersonNamesFormatterOptions {
+    /// the target locale will maximized when passed into the formatter.
     pub target_locale: Locale,
     pub order: FormattingOrder,
     pub length: FormattingLength,
     pub usage: FormattingUsage,
     pub formality: FormattingFormality,
 }
+
+impl PersonNamesFormatterOptions {
+    pub fn new(
+        target_locale: Locale,
+        order: FormattingOrder,
+        length: FormattingLength,
+        usage: FormattingUsage,
+        formality: FormattingFormality,
+    ) -> Self {
+        let lc = LocaleExpander::new();
+        let mut final_locale = target_locale.clone();
+        lc.maximize(&mut final_locale);
```

Comment:
Not sure about running LocaleExpander in this particular spot It needs data which needs to be pluggable.

Reply:
Do we have a way of enforcing/detecting a fully expanded locale ? It's part of the requirement.